[Jianhui] iP

Jaylenlee opened the pull request.

Commit:
Add Gradle support

Jaylenlee committed:
change logo to greeting

Jaylenlee committed:
add echo and exit

Jaylenlee committed:
modularises code

Jaylenlee committed:
added add and list feature

Jaylenlee committed:
add Task class

Jaylenlee committed:
add done command

Jaylenlee committed:
change command function to be inside hashmap

Jaylenlee committed:
add ToDoTask, add @Override to Task toString()

Jaylenlee committed:
add DeadlineTask

Jaylenlee committed:
add EventTask

Jaylenlee committed:
added todo feature

Jaylenlee committed:
add deadline feature

Jaylenlee committed:
add event feature

Jaylenlee committed:
move Task classes to Task Package

Jaylenlee committed:
add Task package import to Duke

Jaylenlee committed:
change UTF Code to actual character

Jaylenlee committed:
add input and change expected output

Jaylenlee committed:
add Task package compile

Jaylenlee committed:
Add Exception Package

Jaylenlee committed:
Add invalid command check

Jaylenlee committed:
add error handle to done command

Jaylenlee committed:
add error handling to todo command

Jaylenlee committed:
add error handling to deadline command

Jaylenlee committed:
add error handling to event command

Jaylenlee committed:
add exception to runtest

Jaylenlee committed:
add deleteCommand add additional error handling to doneCommand

Jaylenlee committed:
add delete test cases

Jaylenlee committed:
Add Overloaded Constructor

Jaylenlee committed:
Add file to /data folder

Jaylenlee committed:
Add load file function

Jaylenlee committed:
Encapsulate command function in CommandProcessor class

Jaylenlee committed:
Change command function to CommandProcessor class

There are serveral command function in Duke class (main class). This causes Duke class to hold multiple responsibilties which may not be ideal.

Encapsulating these command function in CommandProcessor ensures single responsibility principle is adhered. This improves code quality.

Jaylenlee committed:
Add FileToTaskListConverter class

Jaylenlee committed:
Encapsulate file to task list conversion.

Duke class (main class) handles command function, loading file function and file conversion function.

Encapsulating will reduce the responsibility Duke class holds. Improves code quality.

Jaylenlee committed:
Add isDone function to Task

This allows all implementor of Task to be able to check for done status of Task

Jaylenlee committed:
Add create directory and file feature if file does not exists

File is created at <data> folder of <CS2103 IP> folder of Downloads folder in System's Home.

Jaylenlee committed:
Change path from absolute to relative. Add save task function.

Access to path change allows different system other than Windows to access.
Save task function allows data to be loaded into harddisk and resume on next run.

Jaylenlee committed:
Overload Constructor

InvalidActionException now have a constructor with String parameter. Parameter refers to message to be printed.

Jaylenlee committed:
Add DateTimeInvalidFormatException

DateTimeInvalidFormatException is an exception when the input date time format differ from expected.
It extends from InvalidActionException as a form of invalid action.

Jaylenlee committed:
Change time attribute

DeadlineTask now contains LocalDateTime instead of String for time.
String time parameter for constructor now requires a specified format as specified in the DATE_TIME_FORMAT constant. ("dd MMM yyyy kk:mm")

This allows for a more accurate time representation instead of a String variable of anything. Improves program logic.

Jaylenlee committed:
Change time attribute

EventTask now contains LocalDate and LocalTime instead of String for time.
String time parameter for constructor now requires a specified format. ("YYYY-MM-DD HHmm-HHmm")
Date and Time are in "dd MMM yyyy" and "kk:mm" format respectively.

This allows for a more accurate time representation instead of a String variable of anything. Improves program logic.

Jaylenlee committed:
Update test input for date time input changes

Jaylenlee committed:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Duke.java

#Modification:
#             Deletes command functions and imports (CommandProcessor encapsulate them)
#             Remove print line statement in FileToTaskListConvertor when file not found
#             Use CommandProcessor.run() to run inputted commands

Jaylenlee committed:
Fix task description bug error

Task description has a extra " ]" before the action when loading from file.
Action description now extracted using relative position of first " " character in line on file instead of absolute character index.

Jaylenlee committed:
Merge branch 'branch-Level-7' into master

Jaylenlee committed:
Merge branch 'branch-Level-8' into master

Jaylenlee committed:
Add Getters to Task and Change Deadline/EventTask constructors.

Client now can get information on type, description, date and time for Tasks.
Constructors for DeadlineTask and EventTask now throw DateTimeInvalidFormartException if datetime string inputted is not correctly formatted.

Jaylenlee committed:
Add FileCorruptedException

FileCorruptedException is thrown when the date stored in files are not formatted as follows:

type//doneStatus//description of action//date( if applicable)//time( if applicable)

Jaylenlee committed:
Change File Conversion function

FileToTaskListConverter now extracts data from file and build task list only if data stored is formatted in:
type//doneStatus//description of action//date (if applicable)//time (if applicable)

else a FileCorruptedException will be thrown.

Jaylenlee committed:
Add try-catch for FileCorruptedException.

Program now will catch FileCorruptedException and terminates if it catches it.

Jaylenlee committed:
Change package name to lowercase

Jaylenlee committed:
Add TaskList class to handle list of tasks

Jaylenlee committed:
Change List<Task> to TaskList and add save function

Uses a standardised TaskList object instead of List<Task>
Save function added for encapsulation

Jaylenlee committed:
Add a UI Component

UI runs the program

Jaylenlee committed:
Encapsule code and function

Move all initialisation from Duke to UI.
Move remarks printing to TaskList

Jaylenlee committed:
package processing classes

package process includes
UI
FileToTaskConvertor
CommandProcessor

Jaylenlee committed:
Add on to previous commit

package location change

Jaylenlee committed:
Add JUnit5 SetUp

Version: JUnit5.4

Jaylenlee committed:
Add 2 JUnit test class

Test Target:
TaskList
DeadlineTask

Jaylenlee committed:
Add find command function

The find command function must be in the format of "find <keywords". An EmptyActionException message will be printed if no keywords was typed

Jaylenlee committed:
Add META-INF/MANIFEST.MF

Jaylenlee committed:
Add JavaDoc to task package

Jaylenlee committed:
Add JavaDoc to processor package

Jaylenlee committed:
Add JavaDoc to exception package

Jaylenlee committed:
Update test cases for find command

Jaylenlee committed:
Fix Expected Text alignment

Jaylenlee committed:
Remove package name of parent class

Jaylenlee committed:
Merge branch 'branch-A-JavaDoc' into master

Jaylenlee committed:
Merge branch 'branch-A-CodingStandard' into master

Jaylenlee committed:
Merge branch 'branch-Level-9' into master

# Conflicts:
#	src/main/java/processor/CommandProcessor.java

Jaylenlee committed:
Change wildcard imports to single class.

koonweee commented on src/main/java/processor/CommandProcessor.java:
```diff
@@ -0,0 +1,215 @@
+        this.taskList = taskList;
+    }
+
+    private HashMap<String, Consumer<String>> setUpCommandMap() {
```
Great idea to use HashMap to store commands

koonweee commented on src/main/java/processor/CommandProcessor.java:
```diff
@@ -0,0 +1,215 @@
+            } else {
+                action.accept(command);
+            }
+        } catch (DukeException e){
```
Could perhaps add a more descriptive sentence eg. "An error was thrown: " since the user will be seeing the error

Jaylenlee replied:
Noted. Thanks for the suggestion.

koonweee reviewed (commented):
Many good ideas on implementation overall!

Jaylenlee committed:
Merge branch 'add-gradle-support' into master

Jaylenlee committed:
Add checkstyle file

Jaylenlee committed:
Update checkstyle toolVersion

Original: 8.23
Update: 8.29

Jaylenlee committed:
Correct checkstyle errors on main and test folder

RuiFengg commented on src/main/java/task/DeadlineTask.java:
```diff
@@ -0,0 +1,144 @@
+    }
+
+    /**
+     * Constructor to create this object.
```
You might have forgotten to leave an empty line between the description and the parameter section for the constructors 😨

Jaylenlee replied:
Noted. Thanks for pointing that out.

RuiFengg reviewed (commented):
Minor formatting adjustments to be made to follow coding standards.

Jaylenlee committed:
Add JavaFX dependencies

Jaylenlee committed:
Add a line between description and parameter in JavaDoc

Jaylenlee committed:
Add profile images

Jaylenlee committed:
Add JavaFX and FXML component

Components are not integrated with the current Duke's implementation yet.
Components depends on 2 items:
1) Getting greeting string from UI (UI.getGreeting).
2) Duke's getResponse to return a string reply.

Jaylenlee committed:
Set label height to allow full text

Jaylenlee committed:
Downgrade JavaFX version on xmlns

Original: xmlns="http://javafx.com/javafx/11.0.1"
New: xmlns="http://javafx.com/javafx/11"

Jaylenlee committed:
Integrate JavaFX into Duke

Jaylenlee committed:
Merge branch 'branch-Level-10' into master

Jaylenlee committed:
Update mainClassname in application

Previous: "seedu.duke.Duke"
Current: "Launcher"

Jaylenlee committed:
Remove time getter from Task

DeadlineTask and EventTask implements TimeRelated interface for date and time getters
Removed redundant type getter
FileToTaskListConverter get details now from task to save instead of making its own string to save.

Jaylenlee committed:
Add minor correction to JavaDoc and spacing

Jaylenlee committed:
Add assertion

Jaylenlee committed:
Add sort command function

Sort command sorts TaskList by the alphabetical order of the type, followed by the description of task.

Jaylenlee committed:
Add JavaDoc to TimeRelated Interface

Jaylenlee committed:
Merge pull request #1 from Jaylenlee/branch-A-Assertions

Add Assertion

Jaylenlee committed:
Merge branch 'master' of https://github.com/Jaylenlee/ip into master

Jaylenlee committed:
Merge pull request #2 from Jaylenlee/branch-A-CodeQuality

Improve CodeQuality

Jaylenlee committed:
Merge branch 'master' of https://github.com/Jaylenlee/ip into master

Jaylenlee committed:
Merge pull request #3 from Jaylenlee/branch-C-Sort

Add sort command function

Jaylenlee committed:
Merge branch 'master' of https://github.com/Jaylenlee/ip into master

Jaylenlee committed:
Add back getType for comparision

Task type is now a static final variable (ie constant)

Jaylenlee committed:
Modify to fit checkstyle

Jaylenlee committed:
Add window title

Jaylenlee committed:
Add product screenshots

Jaylenlee committed:
Add code reuse citation

Jaylenlee committed:
Add content to README.md

Jaylenlee committed:
Modify Quick Start instruction

Jaylenlee committed:
Debug on spacing

Jaylenlee committed:
Adjust spacing on README

Jaylenlee committed:
Add sub content table list to Features

Jaylenlee committed:
Rename Ui

Jaylenlee committed:
Remove dash line from reply

Jaylenlee committed:
Made displayPicture circle
